### ISSUE TITLE
attributeChanged isn't firing on assignment to certain attributes of derived custom elements

### ISSUE THREAD

**Issue by pshaughn** [servo/servo#25001]
Many WPT tests in the custom-elements/reactions tree fail in a characteristic way that points at a common cause.

In custom-elements/reactions/HTMLButtonElement, a custom element is made that derives from HTMLButtonElement, and its .autofocus is assigned to, using `instance[jsAttributeName]=...` syntax as opposed to `instance.setAttribute(...)`. No attributeChanged event fires. For HTMLButtonElement, it's specifically autofocus that fails like this; for other elements it's various other specific attributes. Searching in test output for "must enqueue an attributeChanged reaction" seems to list them all:

slot on Element
contentEditable on ElementContentEditable
autofocus,  on HTMLButtonElement
rel, referrerPolicy, shape, alt, download, ping, coords,  on HTMLAreaElement
target on HTMLBaseElement
tabIndex, draggable, translate, spellcheck, dir, accessKey,  on HTMLElement
src, type, height, width on HTMLEmbedElement
name on HTMLFieldSetElement
referrerPolicy, sizes, decoding on HTMLImageElement
name on HTMLMapElement
max, value, high, low, min, optimum on HTMLMeterElement
reversed, type, start on HTMLOListElement
httpEquiv on HTMLMetaElement
label on HTMLOptGroupElement
dateTime, cite on HTMLModElement (both in "del use" and "ins use" cases)
name, value on HTMLParamElement
value, max on HTMLProgressElement
cite on HTMLQuoteElement (both "q use" and "blockquote use")
media on HTMLStyleElement
headers, scope, abbr on HTMLTableCellElement
span on HTMLTableColElement

HTMLOutputElement, HTMLSelectElement, and HTMLTableElement are getting slightly different failures I'll look at separately.

**Comment by pshaughn** [comment]
The above list is, in fact, a list of most or all of the IDL attributes that are currently unimplemented in Servo! Most of them are explicitly commented out in the IDL; HTMLImageElement has some just missing.

**Comment by TimvdLippe** [comment]
Looking at https://wpt.fyi/results/html/dom?product=servo many of the failures are for `tabIndex` and `accessKey`. Seems like if we implement those two, that would pass a whole lot of tests.

Specs: https://html.spec.whatwg.org/multipage/interaction.html#the-accesskey-attribute and https://html.spec.whatwg.org/multipage/interaction.html#the-tabindex-attribute

Not sure how difficult these are to implement though.

**Comment by jdm** [comment]
Yeah, we have a policy of avoiding implementing IDL attributes without also implementing the related feature. In this case, implementing tabbed focus is tracked in #3982, and #13460, #15936, and #20075 were three previous attempts at implementing it.

**Comment by arihant2math** [comment]
accessKey seems like a lower-hanging issue. We could expose an option to the embedder, so it can provide a closure to map access keys to shortcuts.

**Comment by mrobinson** [comment]
I've updated the the list in the original description to use checkboxes and have removed now-non-existent entries and added some new one.
